### PR TITLE
implement `devp2p key gen`

### PR DESCRIPTION
**Pull request by int88**
implement part of #7411

### CODE REVIEW

**Comment by int88** [comment]
@mattsse PTAL :)

**Review by mattsse — CHANGES_REQUESTED** [review]
great!

some nits

File: bin/devp2p/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "devp2p"
+version = "0.0.0"
+publish = false
+edition.workspace = true
+license.workspace = true
```
Comment (DaniPopes) [inline bin/devp2p/Cargo.toml:6]: other keys from workspace too? see the other crates

File: bin/devp2p/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+pub mod cli;
+pub mod commands;
+
+fn main() {
+    use cli::Cli;
+
+    let cli = Cli::parse_args();
+
+    if let Err(err) = cli.run() {
```
[the remark this answers is not in the document]
Comment (int88) [reply bin/devp2p/src/main.rs]: Will update when some subcommands need async setup :)

**Comment by int88** [comment]
@mattsse updated!

**Comment by int88** [comment]
friendly ping @mattsse :)

**Review by mattsse — APPROVED** [review]
lgtm,

I thought about this a bit and my main concern rn is that porting all of these commands will be a lot of work that also adds more maintenance effort for a tool that already exists.

I can see the value in it but what like to hear what @Rjected @shekhirin @emhane think

**Review by Rjected — COMMENTED** [review]
> lgtm,
> 
> I thought about this a bit and my main concern rn is that porting all of these commands will be a lot of work that also adds more maintenance effort for a tool that already exists.
> 
> I can see the value in it but what like to hear what @Rjected @shekhirin @emhane think

adding an subset may make sense, at least the commands which are useful in infrastructure or which are otherwise impactful. I don't think we need to reach full feature parity. I don't really like it as a separate command however. We already have the `reth p2p` command which does some isolated things w.r.t p2p, so I think these would fit there. Another issue with separate binaries is that they are not documented in the book or included in the release workflow.

**Comment by emhane** [comment]
agree with @mattsse and @Rjected. I didn't remember that `enr-cli` exists, my bad @int88 . see updated list https://github.com/paradigmxyz/reth/issues/7411#issuecomment-2031215333, I crossed out a bunch of stuff that shouldn't be in reth.

**Pull request closed without merging**